mycqu/enroll.py

```diff
@@ -0,0 +1,252 @@
+from __future__ import annotations
+
+import json
+import re
+from typing import List, Dict, Any, Optional, ClassVar, Tuple
+
+from ._lib_wrapper.dataclass import dataclass
+from .course import Course, CourseDayTime
+
+from requests import Session
+
+from .utils.datetimes import parse_weekday_str, parse_period_str, parse_weeks_str
+
+__all__ = ['EnrollCourseInfo', 'EnrollCourseTimetable', 'EnrollCourseItem']
+
+ENROLLMENT_COURSE_LIST_URL = "https://my.cqu.edu.cn/api/enrollment/enrollment/course-list?selectionSource="
+ENROLLMENT_COURSE_DETAIL_URL = "https://my.cqu.edu.cn/api/enrollment/enrollment/courseDetails/"
+
+
+def get_enroll_list_raw(session: Session, is_major: bool = True) -> Dict[str: List]:
+    """
+
+    从 my.cqu.edu.cn 上获取学生可选课程
+
+    :param session: 登录了统一身份认证（:func:`.auth.login`）并在 mycqu 进行了认证（:func:`.mycqu.access_mycqu`）的 requests 会话
+    :type session: Session
+    :param is_major: 是否查询主修专业可选课程(为false时查询辅修专业可选课程)
+    :type is_major: bool
+    :raises MycquUnauthorized: 若会话未在 my.cqu.edu.cn 进行认证
+    :return: 反序列化获取可选课程的的json
+    :rtype: Dict
+    """
+
+    url = ENROLLMENT_COURSE_LIST_URL + "主修" if is_major else "辅修"
```

Comment:
`+` 的优先级高于 `if`，此处我觉得你想要的是：
```suggestion
    url = ENROLLMENT_COURSE_LIST_URL + ("主修" if is_major else "辅修")
```